onnx/defs/tensor/defs.cc

```diff
@@ -2594,4 +2514,122 @@ ONNX_OPERATOR_SET_SCHEMA(
           }
         }));
 
-} // namespace ONNX_NAMESPACE
+static const char* Pad_ver11_doc = R"DOC(
+Given a tensor containing the data to be padded (`data`), a tensor containing the number of start and end pad values for axis (`pads`), (optionally) a `mode`, and (optionally) `value`, 
+a padded tensor (`output`) is generated.
+
+The three supported `modes` are (similar to corresponding modes supported by `numpy.pad`):
+
+1) `constant`(default) - pads with a given constant value as specified by `value` (which defaults to 0)
+
+2) `reflect` - pads with the reflection of the vector mirrored on the first and last values of the vector along each axis
+
+3) `edge` - pads with the edge values of array
+
+Example (`constant` mode):
+  Insert 0 pads to the beginning of the second dimension.
+
+  data = 
+  [
+      [1.0, 1.2],
+      [2.3, 3.4],
+      [4.5, 5.7],
+  ] 
+
+  pads = [0, 2, 0, 0]
+
+  mode = 'constant'
+
+  value = 0.0
+
+  output = 
+  [
+      [
+          [0.0, 0.0, 1.0, 1.2],
+          [0.0, 0.0, 2.3, 3.4],
+          [0.0, 0.0, 4.5, 5.7],
+      ],
+  ]
+
+)DOC";
+
+ONNX_OPERATOR_SET_SCHEMA(
+    Pad,
+    11,
+    OpSchema()
+        .Attr(
+            "mode",
+            "Supported modes: `constant`(default), `reflect`, `edge`",
+            AttributeProto::STRING,
+            std::string("constant"))
+        .SetDoc(Pad_ver11_doc)
+        .Input(0, "data", "Input tensor.", "T")
+        .Input(
+            1,
+            "pads",
+            "Tensor of integers indicating the number of padding elements to add or remove (if negative) "
+            "at the beginning and end of each axis. For 2D input tensor, it is the number of pixels. "
+            "`pads` should be a 1D tensor of shape [2 * input_rank]. "
+            "`pads` format should be: [x1_begin, x2_begin,...,x1_end, x2_end,...], "
+            "where xi_begin is the number of pad values added at the beginning of axis `i` and "
+            "xi_end, the number of pad values added at the end of axis `i`.",
+            "tensor(int64)")
```

Comment:
I feel `int32` is big enough for `pads`.

Reply:
I prefer to keep it `int64` to be consistent with other usage - 

1) Previously the attributes were `int64` type

2) Other very similar semantics use `int64`. For example, ConstantOfShape - the tensor type for the shape is `int64` (where `int32` would actually do)

3) There is not much value add in making it `int32` (not much space savings) when it is safer to keep it `int64`.

Reply:
Make senses. Thanks.